### PR TITLE
Add possibly to access store from initState

### LINKED ISSUE #155: How to get store in initState life cycle ?
Now, I get store only by StoreConnector, if I want to get store in  initState life cycle, how to do ?

### PR DESCRIPTION
**Pull request by Taym95**
I implemented the solution provided by @brianegan, fixes #155 

I added test but somehow is failing I am skipping the test for now, I debugged it and I could see action comes to `identityReducer` after dispatching it from `initState`, any idea why? thanks!

### CODE REVIEW

File: test/flutter_redux_test.dart [hunk 1]
```diff
@@ -164,6 +164,32 @@ void main() {
       expect(numBuilds, 2);
     });
 
+    testWidgets(
+        'builds the latest state of the store after a change event is triggered from initState',
```
Comment (brianegan) [inline test/flutter_redux_test.dart]: Let's not test this (rebuilds are covered by other test). Instead, let's make a test called `"can access store from initState`. The test should create a `StatefulStoreCaptor` nested inside a `StoreProvider`.

The `StatefulStoreCaptor` should not be given any state, but it should rather capture the store in the state object during `initState`.

It should then check whether the store it captures is the same as the store created at the beginning of this test. You can use a `GlobalKey` to access state variables.

The final check in this test should be `expect(expect(StatefulStoreCaptor.captorKey.currentState.store, store);`

File: test/flutter_redux_test.dart [hunk 2]
```diff
@@ -665,6 +691,33 @@ class StoreCaptor<S> extends StatelessWidget {
   }
 }
 
+// ignore: must_be_immutable
+class StoreCaptorStatefull extends StatefulWidget {
+  static const Key captorKey = Key('StoreCaptorStatefull');
+
+  String state;
```
Comment (brianegan) [inline test/flutter_redux_test.dart]: Not needed here, can remove `ignore: must_be_immutable`

File: test/flutter_redux_test.dart [hunk 3]
```diff
@@ -665,6 +691,33 @@ class StoreCaptor<S> extends StatelessWidget {
   }
 }
 
+// ignore: must_be_immutable
+class StoreCaptorStatefull extends StatefulWidget {
+  static const Key captorKey = Key('StoreCaptorStatefull');
```
Comment (brianegan) [inline test/flutter_redux_test.dart]: Convert this into a globalKey and use it in the constructor

File: test/flutter_redux_test.dart [hunk 4]
```diff
@@ -665,6 +691,33 @@ class StoreCaptor<S> extends StatelessWidget {
...
+
+  String state;
+
+  StoreCaptorStatefull(this.state, {Key key}) : super(key: key);
```
Comment (brianegan) [inline test/flutter_redux_test.dart]: `StoreCaptorStatefull(this.state) : super(key: captorKey);`

File: test/flutter_redux_test.dart [hunk 5]
```diff
@@ -665,6 +691,33 @@ class StoreCaptor<S> extends StatelessWidget {
...
+
+  _StoreCaptorStatefullState createState() => _StoreCaptorStatefullState();
+}
+
+class _StoreCaptorStatefullState extends State<StoreCaptorStatefull> {
```
Comment (brianegan) [inline test/flutter_redux_test.dart]: Add a mutable field `Store<String> store` to the state class

File: test/flutter_redux_test.dart [hunk 6]
```diff
@@ -665,6 +691,33 @@ class StoreCaptor<S> extends StatelessWidget {
...
+  @override
+  void initState() {
+    super.initState();
+    StoreProvider.of<String>(context, listen: false).dispatch('A');
```
Comment (brianegan) [inline test/flutter_redux_test.dart]: Change this to `store = StoreProvider.of<String>(context, listen: false)`

File: lib/flutter_redux.dart [hunk 7]
```diff
@@ -41,10 +41,13 @@ class StoreProvider<S> extends InheritedWidget {
   ///   }
   /// }
   /// ```
-  static Store<S> of<S>(BuildContext context) {
+  static Store<S> of<S>(BuildContext context, {bool listen = true}) {
```
Comment (brianegan) [inline lib/flutter_redux.dart]: Woohoo! Tests look great, but I forgot about one important thing: Documentation!

Could you please add a sentence or two on the `listen` property and why it should be used? Maybe even add a short example of using it in `initState`?

Comment (Taym95) [reply lib/flutter_redux.dart]: I add some sentence and example, what do you think?

**Comment by brianegan** [comment]
Nice work :) Thanks so much for contributing!

**Pull request merged**